src/main/java/frc/robot/vision/multivisionsources/MultiVisionSources.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.vision.multivisionsources;
+
+import frc.constants.VisionConstants;
+import frc.robot.vision.data.VisionData;
+import frc.robot.vision.sources.VisionSource;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class MultiVisionSources<ReturnType extends VisionData> {
+
+	private final String logPath;
+	protected final List<VisionSource<ReturnType>> visionSources;
+
+	@SafeVarargs
+	public MultiVisionSources(String logPath, VisionSource<ReturnType>... visionSources) {
+		this(logPath, List.of(visionSources));
+	}
+
+	public MultiVisionSources(String logPath, List<VisionSource<ReturnType>> visionSources) {
+		this.logPath = logPath;
+		this.visionSources = visionSources;
+	}
+
+	public ArrayList<ReturnType> getUnfilteredVisionData() {
+		return createMappedCopyOfSources(visionSources, VisionSource::getVisionData);
+	}
+
+	public ArrayList<ReturnType> getFilteredVisionData() {
+		return createMappedCopyOfSources(visionSources, VisionSource::getFilteredVisionData);
+	}
+
+	public void periodic() {
+		log();
+	}
+
```

Comment:
move to the end (convention). 
Also, I would delete this function. It causes it to look as it actually logging something but it doesn't too anything by its own. It's better to explicitly run a log func in robotperiodic, rather than debugging why it's not logging anything. 

(I've run into this problem while testing yesterday. I was not aware that it's not a GBSubsystem anymore).

Reply:
This function is now deleted.
> move to the end (convention).

This directly conflicts with [hillel's comment](https://github.com/GreenBlitz/GB-Robot-Template/pull/519#discussion_r1899148124)
also see rant in [667a63f](https://github.com/GreenBlitz/GB-Robot-Template/pull/519/commits/667a63f9d45d5d313c7778270cee78060cb408a6)

Reply:
Whatever the order is. I don’t really care, Hilel knows better the order than I am. 

Still, I don’t think we should have that method.